### DISCUSSION TITLE
How to detect knife round if exists?

### DISCUSSION THREAD

**Discussion by ysomad** [markus-wa/demoinfocs-golang#365] [Q&A] [answered]
Hi! I'm collecting stats from demo and I don't need to store stats from first knife round (for example if a game demo from faceit)

I'm detecting is game started to not collect data from warmup but I didn't find any solution to detect knife rounds. Hope I'll find an answer here!

**Comment by markus-wa** [comment]
Hey!

Knife rounds are generally implemented as what I call "plugin magic" - so there is really no way of detecting it "easily".

The best way to check is probably to see if any of the players have any primary or secondary weapons after freezetime ends.

If yes: it's a real round
If no: ignore the round

**Comment by ysomad** [comment, marked as answer]
Came up with this solution:
```go
for _, p := range gs.TeamCounterTerrorists().Members() {
			weapons := p.Weapons()
			if len(weapons) == 1 && weapons[0].Type == common.EqKnife {
				return
			}
		}
```

I guess it's works, gonna test now, thanks!
